### PR TITLE
feat(chart): Configurable liveness/readiness probes for api

### PR DESCRIPTION
**Pull request by mumoshu**
Ref #268

### CODE REVIEW

**Comment by mumoshu** [comment]
This partially resolves #268 by making probes configurable only for api.
Hope someone replicates it for gw, cr, ctrl, and so on.

**Comment by mumoshu** [comment]
I've verified this to work by running:

```
helm upgrade --install \
  --set api.livenessProbe.initialDelaySeconds=100 \
  --set api.livenessProbe.timeoutSeconds=101 \
  --set api.readinessProbe.periodSeconds=102 \
  ./charts/brigade
```

and then seeing these values are reflected to the pod spec:

```yaml
    # *snip*
    image: deis/brigade-api:v0.9.0
    imagePullPolicy: IfNotPresent
    livenessProbe:
      failureThreshold: 3
      httpGet:
        path: /healthz
        port: 7745
        scheme: HTTP
      initialDelaySeconds: 100
      periodSeconds: 10
      successThreshold: 1
      timeoutSeconds: 101
    name: brigade
    ports:
    - containerPort: 7745
      protocol: TCP
    readinessProbe:
      failureThreshold: 3
      httpGet:
        path: /healthz
        port: 7745
        scheme: HTTP
      periodSeconds: 102
      successThreshold: 1
      timeoutSeconds: 1
    # *snip*
```

And running:

```
helm upgrade --install mybrig1 ./charts/brigade
```

to successfully result in default values:

```yaml
    livenessProbe:
      failureThreshold: 3
      httpGet:
        path: /healthz
        port: 7745
        scheme: HTTP
      periodSeconds: 10
      successThreshold: 1
      timeoutSeconds: 1
    name: brigade
    ports:
    - containerPort: 7745
      protocol: TCP
    readinessProbe:
      failureThreshold: 3
      httpGet:
        path: /healthz
        port: 7745
        scheme: HTTP
      periodSeconds: 10
      successThreshold: 1
      timeoutSeconds: 1
```

File: charts/brigade/templates/api-deployment.yaml [hunk 1]
```diff
@@ -28,10 +28,12 @@ spec:
           httpGet:
             path: /healthz
             port: {{ .Values.api.service.internalPort }}
+{{ if .Values.api.linvenessProbe }}{{ toYaml .Values.api.livenessProbe | indent 10 }}{{ end }}
```
Comment (mumoshu) [inline charts/brigade/templates/api-deployment.yaml]: Note that if you omit the if-end, you end up with the following error when `helm install` is run:

```
Error: UPGRADE FAILED: YAML parse error on brigade/templates/api-deployment.yaml: error converting YAML to JSON: yaml: line 31: could not find expected ':'
```

It is likely to be happening due to the empty `.livenessProbe` key producing `null` or alike, which breaks yaml.

Comment (mumoshu) [reply charts/brigade/templates/api-deployment.yaml]: FYI: A similar implementation in the wild: https://github.com/Azure/helm-charts/blob/4f11be40cc7a47e776faca3fa566eb08c54cfb3b/drupal/templates/deployment.yaml#L81

**Comment by technosophos** [comment]
As always, @mumoshu, thank you for your meticulously written fix. You write exceptionally good code.

**Pull request merged**